Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,326 @@
+                break;
+            case ApiGatewayEmulatorMode.HttpV1:
+            case ApiGatewayEmulatorMode.HttpV2:
+                headers.Add("Apigw-Requestid", GenerateRequestId());
+                break;
+        }
+
+        return headers;
+    }
+
+    /// <summary>
+    /// Generates a random API Gateway ID for REST API mode.
+    /// </summary>
+    /// <returns>A string representing a random API Gateway ID in the format used by API Gateway for REST APIs.</returns>
+    /// <remarks>
+    /// The generated ID is a 12-character string where digits are replaced by letters (A-J), followed by an equals sign.
+    private static string GenerateApiGwId()
+    {
+        return new string(Guid.NewGuid().ToString("N").Take(12).Select(c => char.IsDigit(c) ? (char)(c + 17) : c).ToArray()) + "=";
+    }
+
+
+    /// <summary>
+    /// Generates a random X-Amzn-Trace-Id for REST API mode.
+    /// </summary>
+    /// <returns>A string representing a random X-Amzn-Trace-Id in the format used by API Gateway for REST APIs.</returns>
+    /// <remarks>
+    /// The generated trace ID includes:
+    /// - A root ID with a timestamp and two partial GUIDs
+    /// - A parent ID
+    /// - A sampling decision (always set to 0 in this implementation)
+    /// - A lineage identifier
+    /// </remarks>
+    private static string GenerateTraceId()
+    {
+        var timestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds().ToString("x");
+        var guid1 = Guid.NewGuid().ToString("N");
+        var guid2 = Guid.NewGuid().ToString("N");
+        return $"Root=1-{timestamp}-{guid1.Substring(0, 12)}{guid2.Substring(0, 12)};Parent={Guid.NewGuid().ToString("N").Substring(0, 16)};Sampled=0;Lineage=1:{Guid.NewGuid().ToString("N").Substring(0, 8)}:0";
+    }
+
+
+    /// <summary>
+    /// Generates a random API Gateway request ID for HTTP API v1 and v2.
+    /// </summary>
+    /// <returns>A string representing a random request ID in the format used by API Gateway for HTTP APIs.</returns>
+    /// <remarks>
+    /// The generated ID is a 14-character string consisting of lowercase letters and numbers, followed by an equals sign.
+    private static string GenerateRequestId()
+    {
+        return Guid.NewGuid().ToString("N").Substring(0, 8) + Guid.NewGuid().ToString("N").Substring(0, 6) + "=";
+    }
+
+    /// <summary>
+    /// Sets the response body on the HttpResponse.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the body on.</param>
+    /// <param name="body">The body content.</param>
+    /// <param name="isBase64Encoded">Whether the body is Base64 encoded.</param>
+    private static void SetResponseBody(HttpResponse response, string? body, bool isBase64Encoded, ApiGatewayEmulatorMode apiGatewayEmulator)
+    {
+        if (!string.IsNullOrEmpty(body))
+        {
+            byte[] bodyBytes;
+            if (isBase64Encoded && ApiGatewayEmulatorMode.Rest != apiGatewayEmulator)
```

Comment:
Why not base 64 decided for `Rest`? That seems valid for a Rest Lambda function to return the response base 64 encoded.

Reply:
when i tested with the Rest api gateway - the response doesnt actually get decoded from base 64. it does get decoded when i use httpv1 or httpv2 though

Reply:
Not sure what you mean. It is up to the Lambda function itself to base 64 encode what it sets to the `Body` property of the `APIGatewayProxyResponse`. There is nothing at the service layer that automatically encodes the body response.

Reply:
right. but what i mean was i made a lambda function with 

```
export const handler = async (event, context) => {
    return {
        statusCode: 200,
        body: "dGVzdA==",
        isBase64Encoded: true
    };
};
```

and connected an api gateway to it (both rest and http). when i invoke the api gateway url for the http one, the response is `test`. when i invoke the rest api, the response is `dGVzdA==`. so it seems like in rest api case, api gateway does not automatically decode the response